src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty()) {
+            throw new AuthenticationException("Can not authenticate user");
+        }
+        User user = userFromDbOptional.get();
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(hashedPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Invalid password");
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (!userService.findByEmail(email).isEmpty()) {
+            throw new RegistrationException("User with email " + email + " is already exist");
+        }
+        HashUtil.getSalt();
+        HashUtil.hashPassword(password, HashUtil.getSalt());
+        User user = new User();
+        user.setEmail(email);
+        user.setPassword(password);
+        user.setSalt(user.getSalt());
```

Comment:
This line incorrectly sets the salt. You should set the salt that was generated and used for hashing the password.